proposals/XX-bw-xxx.txt

```diff
@@ -0,0 +1,178 @@
+
+2. Format details
+
+  Bandwidth measurements MUST contain the following sections:
+  - Header (exactly once)
+  - Relays measurements (zero or more times)
+
+2.1. Definitions
+
+  The following nonterminals are defined in dir-spec.txt, sections
+  1.2., 2.1.1., 2.1.3. :
+
+    Int
+    SP (space)
+    NL (newline)
+    Keyword
+    ArgumentChar
+    fingerprint (hexdigest)
+    nickname
+
+  Nonterminals defined in "Tor Directory List Format" (dir-list-spec.txt),
+  section 2.2.1.:
+
+    version_number
+
+  We define the following nonterminals:
+
+    value ::= ArgumentChar+
+    key_value ::= Keyword "=" value
+    line := ArgumentChar* NL
+    timestamp = Int
+    bandwidth = Int
+
+2.2. Header format
+
+It consists of:
+
+  timestamp NL
+
+    [At start, exactly once.]
+
+    The Unix Epoch time in seconds when the file was created.
+
+  "version="version_number NL
+
+    [In second position, zero or one time.]
+
+    The specification document format version.
+
+    New implementations SHOULD include this line.
+    For documents generated following this specification,
+    the version_number MUST be "1.1.0".
+    Documents generated before this specification do not contain this line,
+    and the version_number is considered to be "1.0.0".
+    This line is ignored by any Tor version released before this specification.
+    TODO: implement this feature in tor
+
+  "software="Keyword NL
+
+    [Zero or one time.]
+
+    The name of the software that created the document.
+
+    New implementations SHOULD include this line.
+    Documents generated before this specification do not contain this line,
+    and the software is considered to be "Torflow".
+    This line is ignored by any Tor version released before this specification.
+
+  "software_version="version_number NL
+
+    [Zero or one time.]
+
+    The version of the software that created the document.
+    New implementations SHOULD include this line.
+    Documents generated before this specification do not contain this line.
+    This line is ignored by any Tor version released before this specification.
+
+  Future releases may include additional header lines.
+  Parsers MUST NOT rely on the order of these additional lines.
+  Additional header lines will be accompanied by a minor version increment.
+  If a header line does conform to this format, the line SHOULD be ignored by
+  parsers.
+  This section ends when a relay measurement is found conforming to the next
+  section.
+  TODO: check that it is what tor does
+
+2.3. Relay measurements format
+
+  It consists of zero or more lines that contain the measurement results
+  of relays in arbitrary order.
+  There can be at most one such line per relay identity (fingerprint).
+  Each line MUST consists of the following fields in arbitrary order:
+
+    "node_id=" fingerprint SP
+    "bw=" bandwidth SP
+
+  Where bandwidth is the aggregated measured bandwidth of this relay,
+  in kilobytes per second.
```

Comment:
Pastly said on IRC that we should explain this part better:

The bandwidths need to be on the same scale across all the voting bwauths. Torflow uses kilobytes per second. Other implementations should produce results that are on a similar scale to torflow.